common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -364,18 +364,18 @@ public void record(Object hint) {
         }
 
         private void record0(Object hint, int recordsToSkip) {
-            if (creationRecord != null) {
+            if (creationRecord != null && MAX_RECORDS > 0) {
```

Comment:
@louxiu maybe I miss-read the diff but it seems when `MAX_RECORDS == 0` previous to this change a new record was created and added to lastRecords which is not true anymore. I just want to ensure this is what you want